memory/src/main/scala/filodb.memory/format/BinaryVector.scala

```diff
@@ -343,7 +345,20 @@ trait BinaryAppendableVector[@specialized(Int, Long, Double, Boolean) A] {
   def checkSize(need: Int, have: Int): AddResponse =
     if (!(need <= have)) VectorTooSmall(need, have) else Ack
 
-  @inline final def incNumBytes(inc: Int): Unit = UnsafeUtils.unsafe.getAndAddInt(UnsafeUtils.ZeroPointer, addr, inc)
+  @inline final def incNumBytes(inc: Int): Unit = arch match {
+    case "aarch64"  =>
```

Comment:
Instead of comparing a string in this method, I would compare with a boolean that is set once in a companion class.

Reply:
Good point, I will change it.

Reply:
Nice findings